### ISSUE TITLE
Support of background images

### ISSUE THREAD

**Issue by avstudnitz** [yireo/Yireo_NextGenImages#5]
Just to be sure: background images are not supported, right? We are using code like this in custom widgets:

```
<div class="" style="background-image: url(<?= $escaper->escapeHtmlAttr($image)) ?>);">...</div>
```

We might implement it if it doesn't exist already.

**Comment by avstudnitz** [comment]
Seems JPG fallback support will be tricky then...

**Comment by jissereitsma** [comment]
No, they are not supported. It is tricky anyway. The `picture` element is the main trick as of yet. I have been working myself on a non-Magento site with a PHP script that generates a base64-encoded string for the image and then uses a cookie to switch in between, but this is not ideal either. With Full Page Caching in mind, the solution either requires JavaScript and some cookie-magic (ugly), or a solution like `picture` ...

**Comment by avstudnitz** [comment]
There might be a native solution for that in the future: see https://raoulkramer.de/avif-webp-images-css-background-usage-progressive-enhanced-with-image-set/.

**Comment by jissereitsma** [comment]
Hmm, that's still a loooong wait :(

**Comment by vacla** [comment]
Image sets seemed to be now fully supported.
